### PR TITLE
refactor(tile): convert files to typescript

### PR DESCRIPTION
**Pull request by grabkowski**
### Proposed behaviour

All component files should be refactored and converted to TypeScript.

### Current behaviour

Component files are currently javascript.

### Checklist

- [x] Commits follow our style guide
~~- [ ] Related issues linked in commit messages if required~~
~~- [ ] Screenshots are included in the PR if useful~~
~~- [ ] All themes are supported if required~~
- [x] Unit tests added or updated if required
~~- [ ] Cypress automation tests added or updated if required~~
- [x] Storybook added or updated if required
~~- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required~~
- [x] Typescript `d.ts` file added or updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.

### CODE REVIEW

File: src/components/tile/tile.stories.mdx [hunk 1]
```diff
@@ -202,38 +202,46 @@ Please note that it will not have any effect if the Tile orientation is set to `
 
 <Canvas>
   <Story name="custom Widths">
+    <div>
     <Tile variant="tile" orientation="horizontal" width="75%">
+        <Content title="Test Title">
```
Comment (mkrds) [inline src/components/tile/tile.stories.mdx:207]: It looks like you forgot to move these stories to `tile.stories.tsx` file

**Review by mkrds — APPROVED** [review]

File: src/components/tile/tile.stories.mdx (hunk 1 above)
Comment (mkrds) [reply src/components/tile/tile.stories.mdx:207]: Nevermind, after discussion with Kuba, he showed me that it is impossible to refactor these stories with the current implementation of this component - Kuba will open a separate ticket to fix the buggy implementation

File: src/components/tile/tile.stories.mdx [hunk 2]
```diff
@@ -202,38 +202,46 @@ Please note that it will not have any effect if the Tile orientation is set to `
 
 <Canvas>
   <Story name="custom Widths">
+    <div>
```
Comment (nineteen88) [inline src/components/tile/tile.stories.mdx:205]: Shouldn't we move the stories themselves into a `tile.stories.tsx` file?

Comment (grabkowski) [reply src/components/tile/tile.stories.mdx:205]: @mkrds did already asked for the same thing, please check his second comment:
https://github.com/Sage/carbon/pull/5738#discussion_r1068181986

**Review by nineteen88 — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]

**Pull request merged**